api/gpt.py

```diff
@@ -43,11 +54,33 @@ def add_example(self, ex):
         """Adds an example to the object. Example must be an instance
         of the Example class."""
         assert isinstance(ex, Example), "Please create an Example object."
-        self.examples.append(ex.format())
+        self.examples.append(ex)
+
+    def find_example(self, id):
+        """ Returns the index of the example with specific id"""
+        for i, example in enumerate(self.examples):
+            if example.get_id() == id:
+                return i
```

Comment:
Instead of keeping the examples as a list, change it to a dictionary of `{ id: Example }`. This will simplify the logic for getting and deleting examples